### PR TITLE
doc: Define changelog entries for ISS feature and new attributes

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-262506

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by rubenVB01 — APPROVED** [review]
LGTM

File: .changelog/2420.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ```release-note:note
-resource/mongodbatlas_advanced_cluster: Deprecates following attributes - `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long`.
+resource/mongodbatlas_advanced_cluster: Deprecates following attributes - `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long`. For more details reference the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide).
```
Comment (corryroot) [inline .changelog/2420.txt]:
```suggestion
resource/mongodbatlas_advanced_cluster: Deprecates the `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long` attributes. To learn more, see the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide).
```

File: .changelog/2420.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 ```
 
 ```release-note:note
-data-source/mongodbatlas_advanced_cluster: Deprecates following attributes - `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long`.
+data-source/mongodbatlas_advanced_cluster: Deprecates following attributes - `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long`. For more details reference the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide).
```
Comment (corryroot) [inline .changelog/2420.txt]:
```suggestion
data-source/mongodbatlas_advanced_cluster: Deprecates the `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long` attributes. To learn more, see the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide).
```

File: .changelog/2420.txt [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
 ```
 
 ```release-note:note
-data-source/mongodbatlas_advanced_clusters:Deprecates following attributes - `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long`.
+data-source/mongodbatlas_advanced_clusters: Deprecates following attributes - `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long`. For more details reference the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide).
```
Comment (corryroot) [inline .changelog/2420.txt]:
```suggestion
data-source/mongodbatlas_advanced_clusters: Deprecates the `replication_specs.#.id`, `replication_specs.#.num_shards`, `disk_size_gb`, `advanced_configuration.0.default_read_concern`, and  `advanced_configuration.0.fail_index_key_too_long` attributes. To learn more, see the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide).
```

File: .changelog/2478.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+```release-note:enhancement
+resource/mongodbatlas_advanced_cluster: Support defining cluster shards with independent `replication_specs` objects. This enables defining independent scaled shards. For more details reference the [Advanced Cluster New Sharding Schema Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema).
```
Comment (corryroot) [inline .changelog/2478.txt]:
```suggestion
resource/mongodbatlas_advanced_cluster: Supports defining cluster shards with independent `replication_specs` objects. This feature enables defining independent scaled shards. To learn more, see the [Advanced Cluster New Sharding Schema Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema).
```

File: .changelog/2478.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+```
+
+```release-note:note
+resource/mongodbatlas_advanced_cluster: Making use of this new version will impact the possibility of editing the definition of multi shard clusters in the Atlas UI. This is limited to the first weeks of August.
+```
+
+```release-note:enhancement
+resource/mongodbatlas_advanced_cluster: Adds `replication_specs.*.zone_id` and `replication_specs.*.region_configs.*.(electable_specs|analytics_specs|read_only_specs).disk_size_gb` attributes. For more details reference the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide) and resource documentation.
```
Comment (corryroot) [inline .changelog/2478.txt]:
```suggestion
resource/mongodbatlas_advanced_cluster: Adds `replication_specs.*.zone_id` and `replication_specs.*.region_configs.*.(electable_specs|analytics_specs|read_only_specs).disk_size_gb` attributes. To learn more, see the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide) and resource documentation.
```

File: .changelog/2478.txt [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+```
+
+```release-note:enhancement
+data-source/mongodbatlas_advanced_cluster: Adds `use_replication_spec_per_shard`, `replication_specs.*.zone_id`, and `replication_specs.*.region_configs.*.(electable_specs|analytics_specs|read_only_specs).disk_size_gb` attributes. For more details reference the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide) and data source documentation.
```
Comment (corryroot) [inline .changelog/2478.txt]:
```suggestion
data-source/mongodbatlas_advanced_cluster: Adds `use_replication_spec_per_shard`, `replication_specs.*.zone_id`, and `replication_specs.*.region_configs.*.(electable_specs|analytics_specs|read_only_specs).disk_size_gb` attributes. To learn more, see the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide) and data source documentation.
```

File: .changelog/2478.txt [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+```
+
+```release-note:enhancement
+data-source/mongodbatlas_advanced_clusters: Adds `use_replication_spec_per_shard`, `replication_specs.*.zone_id`, and `replication_specs.*.region_configs.*.(electable_specs|analytics_specs|read_only_specs).disk_size_gb` attributes. For more details reference the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide) and data source documentation.
```
Comment (corryroot) [inline .changelog/2478.txt]:
```suggestion
data-source/mongodbatlas_advanced_clusters: Adds `use_replication_spec_per_shard`, `replication_specs.*.zone_id`, and `replication_specs.*.region_configs.*.(electable_specs|analytics_specs|read_only_specs).disk_size_gb` attributes. To learn more, see the [1.18.0 Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide) and data source documentation.
```

File: .changelog/2478.txt [hunk 8]
```diff
@@ -0,0 +1,19 @@
+```release-note:enhancement
+resource/mongodbatlas_advanced_cluster: Support defining cluster shards with independent `replication_specs` objects. This enables defining independent scaled shards. For more details reference the [Advanced Cluster New Sharding Schema Migration Guide](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema).
+```
+
+```release-note:note
+resource/mongodbatlas_advanced_cluster: Making use of this new version will impact the possibility of editing the definition of multi shard clusters in the Atlas UI. This is limited to the first weeks of August.
```
Comment (corryroot) [inline .changelog/2478.txt]:
```suggestion
resource/mongodbatlas_advanced_cluster: Using this new version impacts the possibility of editing the definition of multi shard clusters in the Atlas UI. This impact is limited to the first weeks of August.
```

**Review by corryroot — APPROVED** [review]
LGTM! I commented with some copy nits.

**Review by oarbusi — APPROVED** [review]

**Pull request merged**